xbmc/interfaces/legacy/WindowInterceptor.h

```diff
@@ -124,9 +124,9 @@ namespace XBMCAddon
 
     public:
       Interceptor(const char* specializedName,
-                  Window* _window, int windowid) : P(windowid, "")
-      { 
-        ((classname = "Interceptor<") += specializedName) += ">";
+                  Window* _window, int windowid) : P(windowid, ""),
+        classname("Interceptor<" + (std::string)specializedName + ">")
```

Comment:
Casting char* to std::string looks weird, I would go for std::string(specializedName)